code/go/pkg/validator/api_test.go

```
@@ -0,0 +1,517 @@
+// packagePath under a single top-level directory (matching elastic-package
+// build output). Returns the path to the zip.
+func createPackageZip(t *testing.T, packagePath string) string {
+	t.Helper()
+	pkgName := filepath.Base(packagePath)
+
+	tmpDir := t.TempDir()
+	zipPath := filepath.Join(tmpDir, pkgName+".zip")
+
+	f, err := os.Create(zipPath)
+	require.NoError(t, err)
+	defer f.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `f.Close` is not checked

(errcheck)